### DISCUSSION TITLE
Converting MLIR/LLVM/RTL format into ONNX.

### DISCUSSION THREAD

**Discussion by salkha21** [onnx/onnx#5941] [Q&A]
Hello Everyone,

I am looking for a tool to convert MLIR code into an ONNX format for my research project. I want to do the reverse of the onnx-mlir tool https://github.com/onnx/onnx-mlir. Does anyone know if there is a tool that can do this or any tips on how to do this?

Thanks.

**Comment by justinchuby** [comment]
Do you know what dialects are used in your MLIR code?

**Reply by salkha21** [reply]
I’m trying to convert it from the affine dialect. If there is any tool also that can convert synthesizable C/C++ into ONNX that would be great as well.
